### PR TITLE
Add ADO `abort-migration` command

### PR DESCRIPTION
**Pull request by begonaguereca**
Hey all! I've added a feature branch so that we can test these all out in smaller chunks. 

I wanted to get this in first so that I can double back and test out the live [API calls.](https://github.com/github/gh-gei/pull/1181)

I'll be opening up a larger PR at the end with all the commands and error scenarios lined up, but for now test out the happy path! 

Add your creds if you haven't done that already: 
```
export GH_PAT="pat"
export ADO_PAT="pat"
```

Kick off a migration: 
```
dotnet run --project src/ado2gh/ado2gh.csproj -- migrate-repo --ado-org valet-testing --ado-team-project build-playground --ado-repo build-playground --github-org import-testing --github-repo abort-repo-testing
```

ABORT it: 
```
dotnet run --project src/ado2gh/ado2gh.csproj -- abort-migration --migration-id RM_kgDaACQyZjk0YzIxYy02ZWFjLTRmZjItYWM3Ni04ZWFkMTBkOWYxOWY
```

- [x ] Did you write/update appropriate tests
- [ ] Release notes updated (if appropriate)
- [ ] Appropriate logging output
- [ ] Issue linked
- [ ] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/OctoshiftCLI.Tests/ado2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+public class AbortMigrationCommandTests
+{
+    [Fact]
+    public void Should_Have_Options()
+    {
+        var command = new AbortMigrationCommand();
+        command.Should().NotBeNull();
+        command.Name.Should().Be("abort-migration");
+        command.Options.Count.Should().Be(1);
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/ado2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs:15]: this seems wrong, it should have 3 options:
`--migration-id`, `--github-pat` and `--verbose`

Comment (begonaguereca) [reply src/OctoshiftCLI.Tests/ado2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs:15]: [c39b523](https://github.com/github/gh-gei/pull/1183/commits/c39b52331c265328c19cf7d26cfc3fb381299f94)

Comment (ArinGhazarian) [reply src/OctoshiftCLI.Tests/ado2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs:15]: options count also needs to be updated to 3

File: src/ado2gh/Commands/AbortMigration/WaitForMigrationCommand.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+using OctoshiftCLI.Commands.AbortMigration;
+
+namespace OctoshiftCLI.AdoToGithub.Commands.WaitForMigration;
+
+public sealed class AbortMigrationCommand : AbortMigrationCommandBase
```
Comment (timrogers) [inline src/ado2gh/Commands/AbortMigration/WaitForMigrationCommand.cs:5]: Hooray for abstraction 🥳

**Pull request merged**